src/lapack/backends/cusolver/cusolver_scope_handle.hpp

```diff
@@ -23,8 +23,10 @@
 #else
 #include <CL/sycl.hpp>
 #endif
-#if __has_include(<sycl/backend/cuda.hpp>)
+#if __has_include(<sycl/context.hpp>)
+#if __SYCL_COMPILER_VERSION <= 20220930
 #include <sycl/backend/cuda.hpp>
```

Comment:
Just checking if we need to have a `has_include` macro for this include file? (It was removed in line 26 above.)

Reply:
Good question! I followed the same changes that had been made to the src/blas/backends/cublas/cublas_scope_handle.hpp file, so we're aligned with the cuBLAS backend.

My understanding is that the location of some of the headers changed based on the compiler version. If there is an overarching CL directory, that's one case. And we can capture that case by checking the existence of sycl/context.hpp (without the CL/ directory), so I don't think we need to verify that sycl/backend/cuda.hpp exists.
If there's not a CL/ directory, then we look at the date of the compiler. Based on #295 , it looks like there were deprecation warnings for the cuda.hpp file, so that's why it's only conditionally included.